The job **`x86_64-gnu-llvm-19-3`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/13805435416/job/38615246869) [(plain)](https://github.com/rust-lang-ci/rust/commit/668bf13c30671f3e951ca591fadb589aaee07f5c/checks/38615246869/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [codegen] tests/codegen/try_question_mark_nop.rs#NINETEEN stdout ----

error in revision `NINETEEN`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-19/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/try_question_mark_nop.NINETEEN/try_question_mark_nop.ll" "/checkout/tests/codegen/try_question_mark_nop.rs" "--check-prefix=CHECK" "--check-prefix" "NINETEEN" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen/try_question_mark_nop.rs:98:36: error: undefined variable: REG3
 // CHECK-NEXT: ret { i64, i64 } [[REG3]]
                                   ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/try_question_mark_nop.NINETEEN/try_question_mark_nop.ll:59:2: note: possible intended match here
 ret { i64, i64 } %3
 ^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen/try_question_mark_nop.NINETEEN/try_question_mark_nop.ll
Check file: /checkout/tests/codegen/try_question_mark_nop.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
           1: ; ModuleID = 'try_question_mark_nop.e67e5ea31bdd6ce8-cgu.0' 
           2: source_filename = "try_question_mark_nop.e67e5ea31bdd6ce8-cgu.0" 
           3: target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128" 
           4: target triple = "x86_64-unknown-linux-gnu" 
           5:  
           6: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
           7: define { i32, i32 } @option_nop_match_32(i32 noundef range(i32 0, 2) %0, i32 %1) unnamed_addr #0 { 
           8: start: 
           9:  %2 = insertvalue { i32, i32 } poison, i32 %0, 0 
          10:  %3 = insertvalue { i32, i32 } %2, i32 %1, 1 
          11:  ret { i32, i32 } %3 
          12: } 
          13:  
          14: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          15: define { i32, i32 } @option_nop_traits_32(i32 noundef range(i32 0, 2) %0, i32 %1) unnamed_addr #0 { 
          16: start: 
          17:  %2 = insertvalue { i32, i32 } poison, i32 %0, 0 
          18:  %3 = insertvalue { i32, i32 } %2, i32 %1, 1 
          19:  ret { i32, i32 } %3 
          20: } 
          21:  
          22: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          23: define { i32, i32 } @result_nop_match_32(i32 noundef range(i32 0, 2) %0, i32 noundef %1) unnamed_addr #0 { 
          24: start: 
          25:  %2 = insertvalue { i32, i32 } poison, i32 %0, 0 
          26:  %3 = insertvalue { i32, i32 } %2, i32 %1, 1 
          27:  ret { i32, i32 } %3 
          28: } 
          29:  
          30: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          31: define { i32, i32 } @result_nop_traits_32(i32 noundef range(i32 0, 2) %0, i32 noundef %1) unnamed_addr #0 { 
          32: start: 
          33:  %2 = insertvalue { i32, i32 } poison, i32 %0, 0 
          34:  %3 = insertvalue { i32, i32 } %2, i32 %1, 1 
          35:  ret { i32, i32 } %3 
          36: } 
          37:  
          38: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          39: define { i32, i32 } @control_flow_nop_match_32(i32 noundef range(i32 0, 2) %0, i32 noundef %1) unnamed_addr #0 { 
          40: start: 
          41:  %2 = insertvalue { i32, i32 } poison, i32 %0, 0 
          42:  %3 = insertvalue { i32, i32 } %2, i32 %1, 1 
          43:  ret { i32, i32 } %3 
          44: } 
          45:  
          46: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          47: define { i32, i32 } @control_flow_nop_traits_32(i32 noundef range(i32 0, 2) %0, i32 noundef %1) unnamed_addr #0 { 
          48: start: 
          49:  %2 = insertvalue { i32, i32 } poison, i32 %0, 0 
          50:  %3 = insertvalue { i32, i32 } %2, i32 %1, 1 
          51:  ret { i32, i32 } %3 
          52: } 
          53:  
          54: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          55: define { i64, i64 } @option_nop_match_64(i64 noundef range(i64 0, 2) %0, i64 %1) unnamed_addr #0 { 
          56: start: 
          57:  %2 = insertvalue { i64, i64 } poison, i64 %0, 0 
          58:  %3 = insertvalue { i64, i64 } %2, i64 %1, 1 
next:98'0                                                 X error: match failed for invalid pattern
next:98'1                                                   undefined variable: REG3
          59:  ret { i64, i64 } %3 
next:98'0     ~~~~~~~~~~~~~~~~~~~~~
next:98'2      ?                    possible intended match
          60: } 
next:98'0     ~~
          61:  
next:98'0     ~
          62: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
next:98'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          63: define { i64, i64 } @option_nop_traits_64(i64 noundef range(i64 0, 2) %0, i64 %1) unnamed_addr #0 { 
next:98'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          64: start: 
          65:  %2 = insertvalue { i64, i64 } poison, i64 %0, 0 
          66:  %3 = insertvalue { i64, i64 } %2, i64 %1, 1 
          67:  ret { i64, i64 } %3 
          68: } 
          69:  
          70: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          71: define { i64, i64 } @result_nop_match_64(i64 noundef range(i64 0, 2) %0, i64 noundef %1) unnamed_addr #0 { 
          72: start: 
          73:  %2 = insertvalue { i64, i64 } poison, i64 %0, 0 
          74:  %3 = insertvalue { i64, i64 } %2, i64 %1, 1 
          75:  ret { i64, i64 } %3 
          76: } 
          77:  
          78: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          79: define { i64, i64 } @result_nop_traits_64(i64 noundef range(i64 0, 2) %0, i64 noundef %1) unnamed_addr #0 { 
          80: start: 
          81:  %2 = insertvalue { i64, i64 } poison, i64 %0, 0 
          82:  %3 = insertvalue { i64, i64 } %2, i64 %1, 1 
          83:  ret { i64, i64 } %3 
          84: } 
          85:  
          86: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          87: define { i64, i64 } @control_flow_nop_match_64(i64 noundef range(i64 0, 2) %0, i64 noundef %1) unnamed_addr #0 { 
          88: start: 
          89:  %2 = insertvalue { i64, i64 } poison, i64 %0, 0 
          90:  %3 = insertvalue { i64, i64 } %2, i64 %1, 1 
          91:  ret { i64, i64 } %3 
          92: } 
          93:  
          94: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
          95: define { i64, i64 } @control_flow_nop_traits_64(i64 noundef range(i64 0, 2) %0, i64 noundef %1) unnamed_addr #0 { 
          96: start: 
          97:  %2 = insertvalue { i64, i64 } poison, i64 %0, 0 
          98:  %3 = insertvalue { i64, i64 } %2, i64 %1, 1 
          99:  ret { i64, i64 } %3 
         100: } 
         101:  
         102: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: write) uwtable 
         103: define void @result_nop_match_128(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([32 x i8]) align 16 dereferenceable(32) %_0, i128 noundef range(i128 0, 2) %0, i128 noundef %1) unnamed_addr #1 { 
         104: start: 
         105:  %2 = getelementptr inbounds i8, ptr %_0, i64 16 
         106:  store i128 %1, ptr %2, align 16 
         107:  store i128 %0, ptr %_0, align 16 
         108:  ret void 
         109: } 
         110:  
         111: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: write) uwtable 
         112: define void @result_nop_traits_128(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([32 x i8]) align 16 dereferenceable(32) %_0, i128 noundef range(i128 0, 2) %0, i128 noundef %1) unnamed_addr #1 { 
         113: start: 
         114:  %2 = getelementptr inbounds i8, ptr %_0, i64 16 
         115:  store i128 %1, ptr %2, align 16 
         116:  store i128 %0, ptr %_0, align 16 
         117:  ret void 
         118: } 
         119:  
         120: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: write) uwtable 
         121: define void @control_flow_nop_match_128(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([32 x i8]) align 16 dereferenceable(32) %_0, i128 noundef range(i128 0, 2) %0, i128 noundef %1) unnamed_addr #1 { 
         122: start: 
         123:  %2 = getelementptr inbounds i8, ptr %_0, i64 16 
         124:  store i128 %1, ptr %2, align 16 
         125:  store i128 %0, ptr %_0, align 16 
         126:  ret void 
         127: } 
         128:  
         129: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: write) uwtable 
         130: define void @control_flow_nop_traits_128(ptr dead_on_unwind noalias nocapture noundef writable writeonly sret([32 x i8]) align 16 dereferenceable(32) %_0, i128 noundef range(i128 0, 2) %0, i128 noundef %1) unnamed_addr #1 { 
         131: start: 
         132:  %2 = getelementptr inbounds i8, ptr %_0, i64 16 
         133:  store i128 %1, ptr %2, align 16 
         134:  store i128 %0, ptr %_0, align 16 
         135:  ret void 
         136: } 
         137:  
         138: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
         139: define { i64, ptr } @result_nop_match_ptr(i64 noundef range(i64 0, 2) %0, ptr noundef %1) unnamed_addr #0 { 
         140: start: 
         141:  %2 = insertvalue { i64, ptr } poison, i64 %0, 0 
         142:  %3 = insertvalue { i64, ptr } %2, ptr %1, 1 
         143:  ret { i64, ptr } %3 
         144: } 
         145:  
         146: ; Function Attrs: mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable 
         147: define { i64, ptr } @result_nop_traits_ptr(i64 noundef range(i64 0, 2) %0, ptr noundef %1) unnamed_addr #0 { 
         148: start: 
         149:  %2 = insertvalue { i64, ptr } poison, i64 %0, 0 
         150:  %3 = insertvalue { i64, ptr } %2, ptr %1, 1 
         151:  ret { i64, ptr } %3 
         152: } 
         153:  
         154: attributes #0 = { mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(none) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
         155: attributes #1 = { mustprogress nofree norecurse nosync nounwind nonlazybind willreturn memory(argmem: write) uwtable "probe-stack"="inline-asm" "target-cpu"="x86-64" } 
         156:  
         157: !llvm.module.flags = !{!0, !1} 
         158: !llvm.ident = !{!2} 
         159:  
         160: !0 = !{i32 8, !"PIC Level", i32 2} 
         161: !1 = !{i32 2, !"RtLibUseGOT", i32 1} 
         162: !2 = !{!"rustc version 1.87.0-nightly (668bf13c3 2025-03-12)"} 
>>>>>>
------------------------------------------



```

</details>

--- a/tests/codegen/try_question_mark_nop.rs
+++ b/tests/codegen/try_question_mark_nop.rs
@@ -63,6 +63,57 @@ pub fn result_nop_traits_32(x: Result<i32, u32>) -> Result<i32, u32> {
     try { x? }
 }
 
+// CHECK-LABEL: @control_flow_nop_match_32
+#[no_mangle]
+pub fn control_flow_nop_match_32(x: ControlFlow<i32, u32>) -> ControlFlow<i32, u32> {
+    // CHECK: start:
+    // CHECK-NEXT: insertvalue { i32, i32 }
+    // CHECK-NEXT: insertvalue { i32, i32 }
+    // CHECK-NEXT: ret { i32, i32 }
+    match x {
+        Continue(x) => Continue(x),
+        Break(x) => Break(x),
+    }
+}
+
+// CHECK-LABEL: @control_flow_nop_traits_32
+#[no_mangle]
+pub fn control_flow_nop_traits_32(x: ControlFlow<i32, u32>) -> ControlFlow<i32, u32> {
+    // CHECK: start:
+    // CHECK-NEXT: insertvalue { i32, i32 }
+    // CHECK-NEXT: insertvalue { i32, i32 }
+    // CHECK-NEXT: ret { i32, i32 }
+    try { x? }
+}
+
+// CHECK-LABEL: @option_nop_match_64
+#[no_mangle]
+pub fn option_nop_match_64(x: Option<u64>) -> Option<u64> {
+    // CHECK: start:
+    // TWENTY-NEXT: %[[TRUNC:[0-9]+]] = trunc nuw i64 %0 to i1
+    // TWENTY-NEXT: %[[SEL:\.[0-9]+]] = select i1 %[[TRUNC]], i64 %1, i64 undef
+    // CHECK-NEXT: [[REG1:%[0-9a-zA-Z_.]+]] = insertvalue { i64, i64 } poison, i64 %0, 0
+    // NINETEEN-NEXT: [[REG2:%[0-9a-zA-Z_.]+]] = insertvalue { i64, i64 } [[REG1]], i64 %1, 1
+    // TWENTY-NEXT: [[REG2:%[0-9a-zA-Z_.]+]] = insertvalue { i64, i64 } [[REG1]], i64 %[[SEL]], 1
+    // CHECK-NEXT: ret { i64, i64 } [[REG2]]
+    match x {
+        Some(x) => Some(x),
+        None => None,
+    }
+}
+
+// CHECK-LABEL: @option_nop_traits_64
+#[no_mangle]
+pub fn option_nop_traits_64(x: Option<u64>) -> Option<u64> {
+    // CHECK: start:
+    // TWENTY-NEXT: %[[TRUNC:[0-9]+]] = trunc nuw i64 %0 to i1
+    // TWENTY-NEXT: %[[SEL:\.[0-9]+]] = select i1 %[[TRUNC]], i64 %1, i64 undef
+    // CHECK-NEXT: insertvalue { i64, i64 }
+    // CHECK-NEXT: insertvalue { i64, i64 }
+    // CHECK-NEXT: ret { i64, i64 }
+    try { x? }
+}
+
 // CHECK-LABEL: @result_nop_match_64
 #[no_mangle]
 pub fn result_nop_match_64(x: Result<i64, u64>) -> Result<i64, u64> {
@@ -86,6 +137,79 @@ pub fn result_nop_traits_64(x: Result<i64, u64>) -> Result<i64, u64> {
     try { x? }
 }
 
+// CHECK-LABEL: @control_flow_nop_match_64
+#[no_mangle]
+pub fn control_flow_nop_match_64(x: ControlFlow<i64, u64>) -> ControlFlow<i64, u64> {
+    // CHECK: start:
+    // CHECK-NEXT: insertvalue { i64, i64 }
+    // CHECK-NEXT: insertvalue { i64, i64 }
+    // CHECK-NEXT: ret { i64, i64 }
+    match x {
+        Continue(x) => Continue(x),
+        Break(x) => Break(x),
+    }
+}
+
+// CHECK-LABEL: @control_flow_nop_traits_64
+#[no_mangle]
+pub fn control_flow_nop_traits_64(x: ControlFlow<i64, u64>) -> ControlFlow<i64, u64> {
+    // CHECK: start:
+    // CHECK-NEXT: insertvalue { i64, i64 }
+    // CHECK-NEXT: insertvalue { i64, i64 }
+    // CHECK-NEXT: ret { i64, i64 }
+    try { x? }
+}
+
+// CHECK-LABEL: @result_nop_match_128
+#[no_mangle]
+pub fn result_nop_match_128(x: Result<i128, u128>) -> Result<i128, u128> {
+    // CHECK: start:
+    // CHECK-NEXT: getelementptr inbounds {{(nuw )?}}i8
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: ret void
+    match x {
+        Ok(x) => Ok(x),
+        Err(x) => Err(x),
+    }
+}
+
+// CHECK-LABEL: @result_nop_traits_128
+#[no_mangle]
+pub fn result_nop_traits_128(x: Result<i128, u128>) -> Result<i128, u128> {
+    // CHECK: start:
+    // CHECK-NEXT: getelementptr inbounds {{(nuw )?}}i8
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: ret void
+    try { x? }
+}
+
+// CHECK-LABEL: @control_flow_nop_match_128
+#[no_mangle]
+pub fn control_flow_nop_match_128(x: ControlFlow<i128, u128>) -> ControlFlow<i128, u128> {
+    // CHECK: start:
+    // CHECK-NEXT: getelementptr inbounds {{(nuw )?}}i8
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: ret void
+    match x {
+        Continue(x) => Continue(x),
+        Break(x) => Break(x),
+    }
+}
+
+// CHECK-LABEL: @control_flow_nop_traits_128
+#[no_mangle]
+pub fn control_flow_nop_traits_128(x: ControlFlow<i128, u128>) -> ControlFlow<i128, u128> {
+    // CHECK: start:
+    // CHECK-NEXT: getelementptr inbounds {{(nuw )?}}i8
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: store i128
+    // CHECK-NEXT: ret void
+    try { x? }
+}
+
 // CHECK-LABEL: @result_nop_match_ptr
 #[no_mangle]
 pub fn result_nop_match_ptr(x: Result<usize, Box<()>>) -> Result<usize, Box<()>> {
@@ -106,28 +230,5 @@ pub fn result_nop_traits_ptr(x: Result<u64, NonNull<()>>) -> Result<u64, NonNull
     // CHECK-NEXT: insertvalue { i{{[0-9]+}}, ptr }
     // CHECK-NEXT: insertvalue { i{{[0-9]+}}, ptr }
     // CHECK-NEXT: ret
-    try { x? }
-}
-
-// CHECK-LABEL: @control_flow_nop_match_32
-#[no_mangle]
-pub fn control_flow_nop_match_32(x: ControlFlow<i32, u32>) -> ControlFlow<i32, u32> {
-    // CHECK: start:
-    // CHECK-NEXT: insertvalue { i32, i32 }
-    // CHECK-NEXT: insertvalue { i32, i32 }
-    // CHECK-NEXT: ret { i32, i32 }
-    match x {
-        Continue(x) => Continue(x),
-        Break(x) => Break(x),
-    }
-}
-
-// CHECK-LABEL: @control_flow_nop_traits_32
-#[no_mangle]
-pub fn control_flow_nop_traits_32(x: ControlFlow<i32, u32>) -> ControlFlow<i32, u32> {
-    // CHECK: start:
-    // CHECK-NEXT: insertvalue { i32, i32 }
-    // CHECK-NEXT: insertvalue { i32, i32 }
-    // CHECK-NEXT: ret { i32, i32 }
     try { x? }
 }

